feat: amount field send all and resizing

fbwoolf opened the pull request:
> Try out this version of the Hiro Wallet - download [extension builds](https://github.com/hirosystems/stacks-wallet-web/actions/runs/4008539053).<!-- Sticky Header Marker -->

This PR fixes (I hope) the amount field font resizing so it works consistently with different symbol lengths, and in both the full page view and extension popup. It also adds the `Send all` functionality to the button in the UI.

![amount-field](https://user-images.githubusercontent.com/6493321/213790466-ba7eb21e-5942-4c6d-a7d1-e987ea24e783.gif)

![amount-field-3](https://user-images.githubusercontent.com/6493321/213790496-80cc24fc-9707-4fa2-bcd1-48fd0d5ba69d.gif)

kyranjamie commented on src/app/pages/send/send-crypto-asset-form/send-form.utils.ts:
```diff
@@ -1,3 +1,44 @@
+  const { balance, fee, fontSize, fontSizeModifier, maxFontSize, pendingTxsBalance } = props;
+  let convertedAmount = '';
+
+  switch (balance.symbol) {
+    case 'BTC':
+      convertedAmount = convertAmountToBaseUnit(balance).minus(fee).toString();
+      break;
+    case 'STX':
+      convertedAmount = convertAmountToBaseUnit(balance)
+        .minus(pendingTxsBalance)
+        .minus(fee)
+        .toString();
+      break;
+    default:
+      convertedAmount = convertAmountToBaseUnit(balance).toString();
+      break;
+  }
```
How can we do this without a switch?

fbwoolf replied:
Yep, I keep using switch statements thinking it is the best way to scale the code if we need to with other blockchains later.

fbwoolf replied:
I get your point, but I did have to remove the `rightInputOverlay` you added to pass in the `SendAllButton` bc the send all button has to trigger resizing the font in the `AmountField`. It made more sense to define `onSendAll` in the amount field component. Also, the fee is only known inside the form and not in the parent to pass in already calculated. I am not sure how to know inside the amount field if I should subtract the fee or not, if I don't use a conditional of some kind? Let me know if you see something I am missing here.

fbwoolf replied:
Nvm, after I wrote that I think I see a way I overlooked.

kyranjamie commented on src/app/pages/send/send-crypto-asset-form/_components/amount-field.tsx:
```diff
@@ -1,57 +1,128 @@
+  const analytics = useAnalytics();
+
+  const { decimals, symbol } = balance;
+  const maxLength = decimals === 0 ? 8 : decimals + 1; // + 1 for decimal char
```
magic number

kyranjamie reviewed (approved):
DIdn't fully understand your reply to my comment, but let's figure out in 1:1 later

fbwoolf commented:
@kyranjamie requesting a new review here bc I refactored after our 1:1. I added back the `rightInputOverlay` and passed in the `SendAllButton` as a prop to the `AmountField` now. I saw some improvements I want to make to the font resizing useEffect but I'll come back to it after I finish up the recipient field changes.

Btw, why call it `rightInputOverlay`? It isn't overlaying anything, right?

kyranjamie commented on src/app/pages/send/send-crypto-asset-form/_components/amount-field.tsx:
```diff
@@ -1,54 +1,88 @@
+
+interface AmountFieldProps {
+  balance: Money;
   rightInputOverlay: JSX.Element;
```
I named this badly, as it's actually positioned at the bottom, so better `bottomInputOverlay`

kyranjamie commented on src/app/pages/send/send-crypto-asset-form/stacks-sip10/stacks-sip10-fungible-token-send-form.tsx:
```diff
@@ -120,8 +130,16 @@ export function StacksSip10FungibleTokenSendForm({
     >
-      <Form>
-        <AmountField symbol={symbol.toUpperCase()} rightInputOverlay={<></>} />
+      <Form style={{ width: '100%' }}>
```
`form` is a block level element, so should be width 100% by default

fbwoolf replied:
I actually had to add this bc during typing in the amount field the form would resize, any other thoughts then on why that would be?

fbwoolf replied:
When it is removed...
![Screen Shot 2023-01-25 at 10 26 57 AM](https://user-images.githubusercontent.com/6493321/214619872-72e3559b-4067-4b48-8c03-1102f52ad3e5.png)

kyranjamie reviewed (approved):
Nice changes @fbwoolf
